cmd/ctr/commands/content/fetch.go

```diff
@@ -66,6 +66,10 @@ Most of this is experimental and there are few leaps to make this work.`,
 			Name:  "all-platforms",
 			Usage: "pull content from all platforms",
 		},
+		cli.BoolFlag{
+			Name:  "all-manifests",
```

Comment:
if the only use of downloading all the manifests is `AppendDistributionSourceLabel`, we can have the flag with a more related name, like `store-distribution-source`.
Otherwise, does it make sense to add distribution source without having all the manifests? If so, they can have independent flags that somehow can be combined.

@dmcgowan what's your recommendation?

Reply:
I don't have strong opinion on this and will hold on for the suggestion of @dmcgowan

Reply:
Flags are tricky, we don't actually support these so it isn't hugely important we get it right the first time. There are a few different "abilities" to keep in mind around pulling: single images vs all images and pulling layers vs shallow pulling metadata. It would be nice if the flag when used on fetch could handle the "shallow" case, that is don't pull tar blobs for any platform but pull all metadata (manifests/configs).

Reply:
Thanks for the feedback! 

So my understanding right now:
1. for `ctr content fetch` when `all-manifests` is used, containerd downloads all manifests/configs but not layers. 
2. for `ctr images pull` when `all-manifests` is used, containerds downloads all manifests/configs + the layers for the matched platform. 

Did I miss anything? If not I will have a new commit to implement the mentioned behavior.